src/aux-server/aux-web/aux-player/scene/PlayerGame.ts

```diff
@@ -1139,23 +1139,77 @@ export class PlayerGame extends Game {
             sim.helper.transaction(asyncResult(e.taskId, null));
         }
     }
-
+    //MY CHANGES START HERE
     private _calculateViewportCoordinatesFromPosition(
         sim: Simulation,
         e: CalculateViewportCoordinatesFromPositionAction
     ) {
-        const portalTag = getPortalTag(e.portal);
+        let portalTag;
+        if (e.portal === 'map') {
+            portalTag = 'mapPortal';
+        } else if (e.portal === 'grid') {
+            portalTag = 'gridPortal';
+        } else if (e.portal === 'miniGrid') {
+            portalTag = 'miniGridPortal';
+        } else if (e.portal === 'miniMap') {
+            portalTag = 'miniMapPortal';
+        } else {
+            portalTag = getPortalTag(e.portal);
+        }
+
         const _3dSim = this._findSimulationForPortalTag(sim, portalTag);
+
         if (_3dSim) {
             const rig = _3dSim.getMainCameraRig();
             const gridScale = _3dSim.getDefaultGridScale();
-            const vector = new Vector3(
-                e.position.x * gridScale,
-                e.position.y * gridScale,
-                e.position.z * gridScale
-            );
+            const isMapPortal =
+                portalTag === 'mapPortal' || portalTag === 'miniMapPortal';
+
+            const position = {
+                x: e.position.x,
+                y: e.position.y,
+                z: e.position.z,
+            };
+
+            let vector;
+
+            if (isMapPortal) {
+                const coordinateTransform = _3dSim.coordinateTransformer
+                    ? _3dSim.coordinateTransformer(position)
+                    : null;
+                if (coordinateTransform) {
+                    vector = new Vector3(0, 0, 0);
+                    vector.applyMatrix4(coordinateTransform);
+                } else {
+                    const lon = position.x * (Math.PI / 180);
+                    const lat = position.y * (Math.PI / 180);
+
+                    const x = EARTH_RADIUS * Math.cos(lat) * Math.cos(lon);
+                    const y = EARTH_RADIUS * Math.sin(lat);
+                    const z = EARTH_RADIUS * Math.cos(lat) * Math.sin(lon);
+
+                    vector = new Vector3(x, y, z);
```

Comment:
@TroyceGowdy Do you know if this code ever gets hit? I believe that coordinateTransform should always be truthy for the map portals, so this section of code presumably does nothing.